packages/backend/src/tokens/request.ts

```diff
@@ -395,3 +485,93 @@ export const debugRequestState = (params: RequestState) => {
   const { isSignedIn, proxyUrl, reason, message, publishableKey, isSatellite, domain } = params;
   return { isSignedIn, proxyUrl, reason, message, publishableKey, isSatellite, domain };
 };
+
+// getActivationEntity determines if the given URL and settings indicate a desire to activate a specific organization or personal workspace.
+function getActivationEntity(url: URL, options: OrganizationSyncOptions): ActivatibleEntity | null {
+  // Check for personal workspace activation
+  if (options.personalWorkspacePatterns) {
+    let matcher: MatchFunction<object>;
+    try {
+      matcher = match(options.personalWorkspacePatterns);
+    } catch (e) {
+      console.error(
```

Comment:
> Let's replace match to use the vendor path-to-regexp introduced https://github.com/clerk/javascript/issues/4145. We'll have to re-export this match function within @clerk/shared.

Done!

> In this case makes more sense to re-thrown to catch invalid patterns during development, since developers might not notice the console.error on the terminal.

Done! Good idea.

c2028328caaf9cc198ea36a14aa42112ebf55df5